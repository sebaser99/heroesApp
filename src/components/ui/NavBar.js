
import { useContext } from 'react';
import { Link, NavLink, useNavigate } from 'react-router-dom';
import { AuthContext } from '../auth/authContext';
import { types } from '../types/types';

export const Navbar = () => {
    const {user} = useContext(AuthContext)
    const navigate = useNavigate()
    const {dispatch} = useContext(AuthContext)

    const handleLogout = ()=>{
        
        const action ={
            type: types.logout,
            payload: {

            }
        }
        dispatch(action)
        navigate('/login', {
            replace: true
        })
    }
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <div className='container-lg'>
                <Link 
                    className="navbar-brand ps-2" 
                    to="/"
                >
                    HeroesAPP
                </Link>

                <div className="navbar-collapse">
                    <div className="navbar-nav">

                        <NavLink 
                            className={({isActive}) => "nav-item nav-link"  + (isActive ? ' active' : '')}
                            to="/marvel"
                        >
                            Marvel
                        </NavLink>

                        <NavLink 
                            className={({isActive}) => "nav-item nav-link"  + (isActive ? ' active' : '')}
                            to="/dc"
                        >
                            DC
                        </NavLink>
                        <NavLink 
                            className={({isActive}) => "nav-item nav-link"  + (isActive ? ' active' : '')}
                            to="/search"
                        >
                            Search
                        </NavLink>
                    </div>
                </div>

                <div className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end">
                    <ul className="navbar-nav ml-auto">
                        <span className='nav-item nav-link text-warning'>{user.name}</span>
                        <button 
                            className="nav-item nav-link btn" 
                            onClick={handleLogout}
                            
                        >
                            Logout
                        </button>
                    </ul>
                </div>

            </div>
        </nav>
    )
}