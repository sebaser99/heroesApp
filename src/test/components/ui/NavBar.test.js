import { mount } from "enzyme"
import { MemoryRouter, Routes, Route } from "react-router-dom"
import { AuthContext } from "../../../components/auth/authContext"
import { types } from "../../../components/types/types";
import { Navbar } from "../../../components/ui/NavBar"

const mockNavigate = jest.fn();

jest.mock('react-router-dom', ()=>({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate
})) 

describe('vamos a probar el <NavBar />', ()=>{
    const contextValue ={
        dispatch : jest.fn(),
        user: {
            name: 'Sebastian'
        }
    }
    const wrapper = mount(
        <AuthContext.Provider value={contextValue} >
            <MemoryRouter initialEntries={['/']}>
                <Routes >
                    <Route path='/' element={<Navbar />} />
                </Routes>
            </MemoryRouter>    
        </AuthContext.Provider>)

    test('debe hacer match con el snapshot', ()=>{
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('.text-info').text().trim()).toBe('Sebastian')
    })

    test('debe llamar el logout', ()=>{
        wrapper.find('button').simulate('click')
        expect(contextValue.dispatch).toHaveBeenCalledWith({'type': types.logout, 'payload': {}})

        expect(mockNavigate).toHaveBeenCalledWith('/login', {"replace": true})
    })
})