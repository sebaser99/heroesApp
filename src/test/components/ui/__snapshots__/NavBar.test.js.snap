// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vamos a probar el <NavBar /> debe hacer match con el snapshot 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <a
    className="navbar-brand ps-2"
    href="/"
    onClick={[Function]}
  >
    Asociaciones
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        className="nav-item nav-link"
        href="/marvel"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        className="nav-item nav-link"
        href="/dc"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        className="nav-item nav-link"
        href="/search"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <div
    className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <span
        className="nav-item nav-link text-info"
      >
        Sebastian
      </span>
      <button
        className="nav-item nav-link btn"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
